src/endpoint.jl

```diff
@@ -58,24 +60,56 @@ function Base.:(==)(a::Endpoint, b::Endpoint)
     a.endpoint == b.endpoint && a.included == b.included
 end
 
+function Base.:(==)(a::LeftEndpoint, b::LeftEndpoint)
+    a.endpoint == b.endpoint && a.included == b.included
+end
+
+function Base.:(==)(a::RightEndpoint, b::RightEndpoint)
+    a.endpoint == b.endpoint && a.included == b.included
+end
+
 function Base.:(==)(a::LeftEndpoint, b::RightEndpoint)
     a.endpoint == b.endpoint && a.included && b.included
 end
 
 function Base.:(==)(a::RightEndpoint, b::LeftEndpoint)
-    a.endpoint == b.endpoint && a.included && b.included
+    b == a
+end
+
+function Base.:(==)(a, b::LeftEndpoint)
+    a == b.endpoint && b.included
+end
+
+function Base.:(==)(a, b::RightEndpoint)
+    a == b.endpoint && b.included
+end
+
+function Base.:(==)(a::LeftEndpoint, b)
+    b == a
+end
+
+function Base.:(==)(a::RightEndpoint, b)
+    b == a
 end
```

Comment:
Can just use `Endpoint` here:
```julia
function Base.:(==)(a, b::Endpoint)
    a == b.endpoint && b.included
end

function Base.:(==)(a::Endpoint, b)
    b == a
end
```
I assume the one-liner versions of these look out of place?

Reply:
These should probably be moved to the end with the `isless(::Any, ::Endpoint)` definitions. Since the logic behind scalar-endpoint and endpoint-endpoint comparisons is rather different I think it makes sense to group by this way instead of by operator.

Reply:
Right, that makes sense. I suppose some of these could be one liners, but then the question is if we want it to be consistently long versions for the entire thing, or mixed, or make them all one liners. Some are little long though